quickwit-index-config/src/config.rs

```diff
@@ -72,12 +82,48 @@ pub enum SortBy {
 
 /// Convert a field (name, value) into a tag string `name:value`.
 pub fn convert_tag_to_string(field_name: &str, field_value: &Value) -> String {
-    format!("{}:{}", field_name, tantivy_value_to_string(field_value))
+    let mut field_value_str = tantivy_value_to_string(field_value);
+    if field_value_str == MANY_TAG_VALUES {
+        field_value_str = format!("{}{}", TAGS_VALUE_ESCAPE, field_value_str);
+    }
+    format!(
+        "{}{}{}",
+        field_name, TAG_FIELD_VALUE_SEPARATOR, field_value_str
+    )
 }
 
 /// Returns true if tag_string is of form `{field_name}:any_value`.
 pub fn match_tag_field_name(field_name: &str, tag_string: &str) -> bool {
-    tag_string.starts_with(&format!("{}:", field_name))
+    tag_string.starts_with(&format!("{}{}", field_name, TAG_FIELD_VALUE_SEPARATOR))
+}
+
+/// Extracts the field name from a tag value of the form `{field_name}:any_value`.
+/// It will return the whole string if the separator is not present (to discuss: should we error?).
+pub fn extract_field_name_from_tag_value(tag_value: &str) -> String {
```

Comment:
why not return `None` if it does not match? I found quite normal to return an option here.

Reply:
The implication I wanted to clear may be more than this return type. So right now we have not been enforcing tags to have this format `{field}:{value}` and this current implementation follows that by **trying** at best to parse if not successful we just return the whole string. Now if we return option/error it means we may need to make sure to error out if the format of tag is not correct.
I still think the correct semantic is what you are hinting for so I will do that.